src/main/java/org/broadinstitute/hellbender/utils/config/ConfigUtils.java

```diff
@@ -0,0 +1,332 @@
+     * @param configFileName The name of the file from which to initialize the configuration
+     * @param configClass The type of configuration in which to interpret the given {@code configFileName}
+     * @return The configuration instance implementing {@link GATKConfig} containing any overrides in the given file.
+     */
+    public static final <T extends Config> T initializeConfigurationAsProperties(final String configFileName, Class<? extends T> configClass) {
+
+        Utils.nonNull(configClass);
+
+        // Get a place to store our properties:
+        final Properties userConfigFileProperties = new Properties();
+
+        // Try to get the config from the specified file:
+        if ( configFileName != null ) {
+
+            try {
+                final FileInputStream userConfigFileInputStream = new FileInputStream(configFileName);
+                userConfigFileProperties.load(userConfigFileInputStream);
+
+                if (configFileName != null) {
+                    System.out.println("Found " + configClass.getSimpleName() + " Configuration File: " + configFileName);
+                }
+
+            } catch (final FileNotFoundException e) {
+                System.err.println("WARNING: unable to find specified " + configClass.getSimpleName() + " configuration file: "
+                        + configFileName + " - defaulting to built-in config settings.");
+            }
+            catch (final IOException e) {
+                System.err.println("WARNING: unable to load specified " + configClass.getSimpleName() + " configuration file: "
+                        + configFileName + " - defaulting to built-in config settings.");
+            }
+        }
+
+        // Cache and return our configuration:
+        // NOTE: The configuration will be stored in the ConfigCache under the key GATKConfig.class.
+        //       This means that any future call to getOrCreate for this GATKConfig.class will return
+        //       Not only the configuration itself, but also the overrides as specified in userConfigFileProperties
+        return ConfigCache.getOrCreate(configClass, userConfigFileProperties);
+    }
+
+    /**
+     * Injects system properties from the given configuration file.
+     * @param config The {@link GATKConfig} object from which to inject system properties.
+     */
+    public static final <T extends Config> void injectSystemPropertiesFromConfig(T config) {
+        injectSystemPropertiesFromConfig(config, "");
+    }
+
+    /**
+     * Injects system properties from the given configuration file.
+     * @param config The {@link GATKConfig} object from which to inject system properties.
+     * @param propertyPrefix Prefix that is required for a property in {@code config} to be imported to System properties.
+     */
+    public static final <T extends Config> void injectSystemPropertiesFromConfig(T config, String propertyPrefix) {
```

Comment:
I think we should split this into two methods for testing, one that gets the system properties that should be tested and a seperate skinny one that does the injection